add libphonenumber w/ git autoupdate

kennynaoh opened the pull request:
PR for #7805 
@x09326  please help me review this pull request, thank you.
The issue discussed with author is below
https://github.com/googlei18n/libphonenumber/issues/1325, https://github.com/googlei18n/libphonenumber/issues/1362

Manually add other versions and change version name from git repo because the old versions come with unusual prefix.
Update .gitignore to ignore the old version with unusual prefix

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.
# Profile of the lib
- Git repository (required): https://github.com/googlei18n/libphonenumber
- Official website (optional, not the repository):
- NPM package url (optional):
- GitHub / Bitbucket popularity (required):
  - Count of stars: 3973
  - Count of watchers: 205
  - Count of forks: 495
- NPM download stats (optional):
  - Downloads in the last day:
  - Downloads in the last week:
  - Downloads in the last month:
# Essential checklist
- [ ] I'm the author of this library
  - [ ] I would like to add link to the page of this library on CDNJS on website / readme
- [x] This lib was not found on cdnjs repo
- [x] No already exist / duplicated issue and PR
- [x] The lib has notable popularity
  - [x] More than 100 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
  - [ ] More than 500 downloads stats per month on npm registry
- [ ] Project has public repository on famous online hosting platform (or been hosted on npm)
# Auto-update checklist
- [x] Has valid tags for each versions (for git auto-update)
- [x] Auto-update setup
- [x] Auto-update target/source is valid.
- [x] Auto-update filemap is correct.
# Git commit checklist
- [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
- [x] The parent of the commit(s) in the PR is not old than 3 days.
- [x] Pull request is sending from a non-master branch with meaningful name.
- [x] Separate unrelated changes into different commits.
- [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
- [x] Close corresponding issue in commit message
- [x] Mention related issue(s), people in commit message, comment.

git auto-update config added, close #7805, cc @lararennie

pvnr0082t commented on ajax/libs/libphonenumber/package.json:
```diff
@@ -0,0 +1,28 @@
+      "metadata!(fortesting*|lite*)",
+      "phonemetadata.pb*",
+      "phonenumber.pb*",
+      "phonenumberutil!(_test*)"
```
I think `"+(asyoutypeformatter|metadata|phonemetadata.pb|phonenumber.pb|phonenumberutil).+(js|map)"` would be easier and cleaner. What do you think? @kennynaoh

PeterDaveHello replied:
This is a little hard to read.

pvnr0082t replied:
So is the config written by @kennynaoh fine?

PeterDaveHello replied:
If it works, it'll be fine, I didn't test it yet.

x09326 commented:
@kennynaoh 
Please rebase to the latest branch.

kennynaoh commented:
@pvnr0082t , @x09326 
I have rebased to the latest master branch.
Please help me review it again.
Thanks!

pvnr0082t commented:
LGTM

x09326 commented:
LGTM

PeterDaveHello commented:
@pvnr0082t @x09326 just to make sure, did you guys test the auto-update config?

pvnr0082t commented:
Yes, I tried it.

kennynaoh commented:
@pvnr0082t , @x09326 
I have changed the version name from `libphonenumber-x.x.x` to `x.x.x` and also added the `.gitignore`
Please help me review it again.
Thanks!

pvnr0082t commented:
@kennynaoh You didn't minify the files in old versions.

pvnr0082t commented on .gitignore:
```diff
@@ -156,6 +156,7 @@ ajax/libs/ng-pdfviewer/r0.2.1/
 ajax/libs/prettyCheckable/.2.1.1/
 ajax/libs/Hyphenator/HyphenatorBeta12/
 ajax/libs/Hyphenator/ersion_*
+ajax/libs/libphonenumber/libphonenumber-*/
```
@PeterDaveHello If this line is added, I think this library will not automatically update. Is it need to discuss with author to change the tagged scheme?

PeterDaveHello replied:
What is the reason why it will not update? All the new versions come with that prefix? If so, yes, I'll like to discuss with its author.

kennynaoh replied:
SURE

kennynaoh commented:
@pvnr0082t , @x09326 
The author published the latest version with non-prefix name.
Please help me review it again.
Thanks!

pvnr0082t commented:
LGTM

PeterDaveHello commented on ajax/libs/libphonenumber/package.json:
```diff
@@ -0,0 +1,32 @@
+          "metadata!(fortesting*|lite*)",
+          "phonemetadata.pb*",
+          "phonenumber.pb*",
+          "phonenumberutil!(_test*)"
```
Let's combine to something like `phone*!(_test*).+(js|map)`?

kennynaoh replied:
This `fileMap` will still grab the file `phonenumberutil_test.js`, maybe we should separate it?

PeterDaveHello replied:
I just say "something like" ...

kennynaoh replied:
I think I can combined `phonemetadata.pb*` and `phonenumber.pb*` to `+(phonemetadata.pb*|phonenumber.pb*)`, and the others remain unchanged.
What about this?

PeterDaveHello replied:
The part I want to reuse here is `phone`.

kennynaoh commented on ajax/libs/libphonenumber/package.json:
```diff
@@ -0,0 +1,32 @@
+        "basePath": "javascript/i18n/phonenumbers",
+        "files": [
+          "asyoutypeformatter!(_test*)",
+          "metadata!(fortesting*|lite*)",
```
OK, I will add it.

kennynaoh commented:
@x09326 @pvnr0082t I have updated the fileMap, please help me review it again. Thanks!

pvnr0082t commented on .gitignore:
```diff
@@ -171,6 +171,159 @@ node_modules/
+ajax/libs/origamijs/0.0.2/
+ajax/libs/mobx-react/*-observableprops*/
+ajax/libs/mobx-react/*-fix*/
+ajax/libs/libphonenumber*/
```
Why did you make so much unrelated changes?

kennynaoh commented:
@pvnr0082t @x09326 I have updated the `.gitignore` file. Please help me review it. Thanks!

pvnr0082t commented on .gitignore:
```diff
@@ -171,6 +171,7 @@ node_modules/
 /ajax/libs/react-data-grid/*-*/
 !ajax/libs/react-data-grid/*-alpha*/
 /ajax/libs/react-data-grid/*-dm*/
+/ajax/libs/libphonenumber*/
```
@kennynaoh Shouldn't it be `/ajax/libs/libphonenumber/libphonenumber*`?

kennynaoh replied:
Right, it's my mistake. Please help me review it. Thanks!

pvnr0082t commented:
LGTM

PeterDaveHello commented:
Ping @x09326

maruilian11 commented on ajax/libs/libphonenumber/package.json:
```diff
@@ -0,0 +1,30 @@
+        "files": [
+          "asyoutypeformatter!(_test*)",
+          "metadata!(fortesting*)",
+          "phone!(*_test*)"
```
@kennynaoh could you tell me why use `!(_test*)` for `asyoutypeformatter`, but `!(*_test*)` for `phone`.

kennynaoh replied:
It's different case. BTW, `!(test*)` is wrong rule, but I think `!(*test*)` it's ok.

kennynaoh commented:
@x09326 @pvnr0082t @maruilian11 I changed `!(*_test*)` to `!(*test*)`. Please help me review it again. Thanks!

pvnr0082t commented:
LGTM

maruilian11 commented:
@kennynaoh Did you change to `!(*test*)`? I saw the old version.

maruilian11 commented on ajax/libs/libphonenumber/package.json:
```diff
@@ -22,7 +22,7 @@
           "asyoutypeformatter!(_test*)",
           "metadata!(fortesting*)",
-          "phone!(*_test*)"
+          "phone!(*test*)"
```
why change it in this commit? not in the first one?

maruilian11 replied:
btw, I think we can use !(*test*) in 3 rules, it's easy to read, and I tested it.

kennynaoh commented:
@x09326 , @pvnr0082t , @maruilian11 I have updated the `fileMap`, please help me review it. Thanks!

pvnr0082t commented:
LGTM

maruilian11 commented:
@kennynaoh I think you did a great job. There is only some things you may notice.
1. 
>The first line of commit message is less then 50 chars, be clean and clear, easy to understand.

I think
`add old versions of libphonenumber and update .gitignore, cc #7805`

`Manually add other versions and change version name from git repo because the old versions come
with unusual prefix.`
`update .gitignore to ignore the old version come with unusual prefix`
can turn to 
`Add libphonenumber old verisons, cc #7805`

`Manually add old versions and change version names because the they come with unusual prefix.`
`Update .gitignore to ignore the old versions with unusual prefix`

2. Don't add v7.7.5 and v7.7.4 to reduce reviewers' loading.
Adding those versions doesn't match the commit msg.

3. not modify the version name, like v7.7.5 -> 7.7.5

Thank you!

kennynaoh commented:
@maruilian11 Thank you for advice

kennynaoh commented:
@maruilian11 @x09326 @pvnr0082t I have updated the commit message and removed two versions `v7.7.4` and `v7.7.5`. Please help me review it. Thanks!

pvnr0082t commented:
LGTM

maruilian11 commented:
@kennynaoh sorry for the late reply.
please rebase this branch with the latest master.
since you need to rebase this branch, can you:
1. change `0.0.8` -> `v0.0.8`
2. let first line of commit message be less then 50 chars

thanks!

kennynaoh commented:
@x09326 @pvnr0082t @maruilian11 My first line of first commit is less than 50 chars and I have rebased to the latest master branch. Please help me review it again. Thanks!

pvnr0082t commented:
LGTM

maruilian11 commented:
LGTM

PeterDaveHello commented on ajax/libs/libphonenumber/package.json:
```diff
@@ -0,0 +1,30 @@
+        "files": [
+          "asyoutypeformatter!(*test*)",
+          "metadata!(*test*)",
+          "phone!(*test*)"
```
Can we just use `!(*+(demo|test)*).js` here?

kennynaoh replied:
That's great!

kennynaoh commented:
@x09326 @pvnr0082t @maruilian11 I have updated fileMap. Please help me review it. Thanks!

kennynaoh commented:
@x09326 @pvnr0082t @maruilian11 I separated the commit to make it more clear. Please help me review it. Thanks!

pvnr0082t commented on .gitignore:
```diff
@@ -187,6 +187,8 @@ node_modules/
 /ajax/libs/jquery-easing/1.3.0/
 /ajax/libs/jquery-easing/1.3.1/
+/ajax/libs/libphonenumber/libphonenumber*/
+
```
unnecessary new line.

kennynaoh commented:
Sorry for that mistake. @x09326 @pvnr0082t @maruilian11 Please help me review it again. Thanks!

x09326 commented:
LGTM
ping @PeterDaveHello

PeterDaveHello commented:
Thank you all.
